packages/kbn-test/src/es/es_test_cluster.js

```diff
@@ -61,10 +62,15 @@ export function createEsTestCluster(options = {}) {
     }
 
     async start(esArgs = []) {
-      const { installPath } =
-        esFrom === 'source'
-          ? await cluster.installSource(config)
-          : await cluster.installSnapshot(config);
+      let installPath;
```

Comment:
While this does work - I believe there could be confusion as there are now two options which provide similar functionality; `esFrom` and `es-install-dir`. I believe it makes sense to also support `esFrom` accepting an absolute path to an ES install.

You can use [path.isAbsolute](https://nodejs.org/api/path.html#path_path_isabsolute_path).

Something like:

```
if (esFrom === 'source') {
  installPath = (await cluster.installSource(config)).installPath;
} else if (esFrom === 'snapshot') {
  installPath = (await cluster.installSnapshot(config)).installPath;
} else if (path.isAbsolute(esFrom)) {
  installPath = esFrom;
} else {
  throw new Error(`unknown option esFrom "${esFrom}"`);
}

Doing this should remove most of the lines from this PR.

Reply:
Updated so path is part of esFrom option.